dynamic_recalibration.py

```diff
@@ -0,0 +1,352 @@
+parser.add_argument("-fps", "--fps", default=10, type=int,
+                    required=False, help="Set capture FPS for all cameras. Default: %(default)s")
+parser.add_argument("-d", "--debug", default=False, action="store_true", help="Enable debug logs.")
+parser.add_argument("-dr", "--dryRun", default=False, action="store_true", help="Dry run, don't flash obtained calib data, just save to disk.")
+options = parser.parse_args()
+
+#TODO implement -brd, -slr
+
+epipolar_threshold = options.maxEpiploarError
+rgbEnabled = not options.disableRgb
+dryRun = options.dryRun
+
+def calculate_Rt_from_frames(frame1,frame2,k1,k2,d1,d2):
+
+    sift = cv2.SIFT_create()
+    kp1, des1 = sift.detectAndCompute(frame1,None)
+    kp2, des2 = sift.detectAndCompute(frame2,None)
+
+    FLANN_INDEX_KDTREE = 1
+    index_params = dict(algorithm = FLANN_INDEX_KDTREE, trees = 5)
+    search_params = dict(checks=50)
+    flann = cv2.FlannBasedMatcher(index_params,search_params)
+    matches = flann.knnMatch(des1,des2,k=2)
+    pts1 = []
+    pts2 = []
+    for i,(m,n) in enumerate(matches):
+        if m.distance < 0.8*n.distance:
+            pts2.append(kp2[m.trainIdx].pt)
+            pts1.append(kp1[m.queryIdx].pt)
+
+    minKeypoints = 20
+    if len(pts1) < minKeypoints:
+        raise Exception(f'Need at least {minKeypoints} keypoints!')
+
+    # img=cv2.drawKeypoints(frame1, kp1, frame1, flags=cv2.DRAW_MATCHES_FLAGS_DRAW_RICH_KEYPOINTS)
+    # cv2.imshow("Left", img)
+    # img2=cv2.drawKeypoints(frame2, kp2, frame2, flags=cv2.DRAW_MATCHES_FLAGS_DRAW_RICH_KEYPOINTS)
+    # cv2.imshow("Right", img2)
+
+    pts1 = np.float32(pts1)
+    pts2 = np.float32(pts2)
+    E, mask = cv2.findEssentialMat(pts1,pts2,k1,d1,k2,d2)
+    # print(f"essential_matrix: {E}")
+
+    points, R_est, t_est, mask_pose = cv2.recoverPose(E, pts1,pts2, mask=mask)
+    # print(frame1.shape[::-1])
+    flags = 0
+    # flags=cv2.CALIB_ZERO_DISPARITY
+    R1, R2, P1, P2, Q, roi_left, roi_right = cv2.stereoRectify(k1, d1, k2, d2, frame2.shape[::-1], R_est, t_est, flags=flags)
+
+    return R_est, t_est, R1, R2, P1, P2, Q
+
+def calculate_epipolar_error(frame1, frame2):
+    sift = cv2.SIFT_create()
+    kp1, des1 = sift.detectAndCompute(frame1,None)
+    kp2, des2 = sift.detectAndCompute(frame2,None)
+
+    FLANN_INDEX_KDTREE = 1
+    index_params = dict(algorithm = FLANN_INDEX_KDTREE, trees = 5)
+    search_params = dict(checks=50)
+    flann = cv2.FlannBasedMatcher(index_params,search_params)
+    matches = flann.knnMatch(des1,des2,k=2)
+    pts1 = []
+    pts2 = []
+    for i,(m,n) in enumerate(matches):
+        if m.distance < 0.8*n.distance:
+            pts2.append(kp2[m.trainIdx].pt)
+            pts1.append(kp1[m.queryIdx].pt)
+
+    pts1 = np.float32(pts1)
+    pts2 = np.float32(pts2)
+    # this is just to get inliers
+    M, mask = cv2.findHomography(pts1, pts2, cv2.RANSAC,5.0) #TODO try and compare findHomography against recoverPose and extract rotation from obtained homography
```

Comment:
i suggest to use `USAC` -> https://opencv.org/evaluating-opencvs-new-ransacs/

Reply:
Thanks, I switched to `USAC_MAGSAC` for the essential matrix too.